nbgrader/auth/hubauth.py

```diff
@@ -33,17 +35,31 @@ def _hubapi_address_default(self):
         Relative path of the formgrader with respect to the hub's user base
         directory.  No trailing slash. i.e. "Documents" or "Documents/notebooks". """)
     
-    hub_base_url = Unicode(None, config=True, help="Base URL of the hub server.")
+    hub_base_url = Unicode(config=True, help="Base URL of the hub server.")
     def _hub_base_url_default(self):
         return 'http://{}:{}'.format(self.hub_address, self.hub_port)
+    
+    generate_hubapi_token = Bool(False, config=True, help="""Use `jupyterhub token` as a default
+        for HubAuth.hubapi_token instead of $JPY_API_TOKEN.""")
+
+    hubapi_token = Unicode(config=True, help="""JupyterHub API auth token.  
+        Generated by running `jupyterhub token`.  If not explicitly set,
+        nbgrader will use $JPY_API_TOKEN as the API token.""")
+    def _hubapi_token_default(self):
+        if self.generate_hubapi_token:
+            return check_output(['jupyterhub', 'token']).decode('utf-8').strip()
```

Comment:
Erg yes, forgot about that.  It complicates things, I think I'll add a traitlet to allow the sqlite path to be specified instead of the command's CWD.